uberfire-commons-editor/uberfire-commons-editor-api/src/main/java/org/uberfire/ext/editor/commons/service/CopyService.java

```diff
@@ -15,10 +15,32 @@
  */
 package org.uberfire.ext.editor.commons.service;
 
+import java.util.Collection;
+
 import org.jboss.errai.bus.server.annotations.Remote;
+import org.uberfire.backend.vfs.Path;
 import org.uberfire.ext.editor.commons.service.support.SupportsCopy;
 
+/**
+ * Responsible for paths copy.
+ */
 @Remote
 public interface CopyService extends SupportsCopy {
 
+    /**
+     * Copies (in batch) the paths passed in {@param paths}, if they exist.
+     * @param paths Paths that will be removed.
+     * @param newName New path's name.
+     * @param comment Comment about the renaming.
+     */
+    void copyIfExists( final Collection<Path> paths,
+                       final String newName,
+                       final String comment );
+
+    /**
+     * Verifies if a path can be copied.
+     * @param path Path to be verified.
+     * @return true if the path can be copied and false otherwise.
```

Comment:
This looks a bit weird considering the method name. If the method returns `true`, the name suggest is has restriction, so I would assume it should not be copied?